src/daemon/protocols/cdp.c

```diff
@@ -522,6 +583,29 @@ cdp_decode(struct lldpd *cfg, char *frame, int s,
 			TAILQ_INSERT_TAIL(&port->p_vlans,
 					  vlan, v_entries);
 			break;
+#endif
+#ifdef ENABLE_DOT3
+		case CDP_TLV_POWER_AVAILABLE:
+			CHECK_TLV_SIZE(12, "Power Available");
+			/* check if it is a respone to a request id */
+			if (PEEK_UINT16 > 0) {
+				port->p_cdp_power.management_id = PEEK_UINT16;
+				port->p_power.allocated = PEEK_UINT32;
+				port->p_power.allocated /= 100;
+				port->p_power.supported = 1;
```

Comment:
I should perhaps remove p_power.support and p_power.enabled since it is incorrect and irrelevant?